core/src/main/java/google/registry/model/CacheUtils.java

```diff
@@ -73,4 +77,29 @@ public static Caffeine<Object, Object> newCacheBuilder(Duration expireAfterWrite
     }
     return caffeine;
   }
+
+  /**
+   * A {@link CacheLoader} that automatically masquerade the background thread where the refresh
+   * action runs in to be an GAE thread.
+   */
+  @DeleteAfterMigration
+  public abstract static class AppEngineEnvironmentCacheLoader<K, V> implements CacheLoader<K, V> {
```

Comment:
Will any of this be necessary once we get rid of Ofy?

Also, why is this only necessary for some caches, rather than all of them?

Reply:
None of these would be necessary, that's why this class is annotated with `@DeleteAfterMigration` (granted, this annotation is probably overloaded, it could mean after migration to SQL, which is already completed, or after migration off of Ofy).

I've actually changed all usages of `CacheLoader`, except for one used in testing only. This might already be an overkill as some of the `CacheLoader` usages do not require the creation of a `VKey`.

https://cs.opensource.google/search?q=cacheloader&sq=&ss=nomulus%2Fnomulus

Reply:
So, considering that either way we go is going to be reverted after the imminent deactivation of Datastore stuff, why not just go with PR #1638, which is simpler?

Reply:
Simpler and less risky.

Reply:
With this PR we do get the async refresh feature, which was the main reason to switch to caffeine. Plus this fix is not as intrusive as originally thought. Also we don't need a bug or a TODO to remind us to turn back on async refresh -- we will delete the annotated classes, and then we will have to rename anonymous `AppEngineEnvironmentCacheLoader` back to `CacheLoader`.

So it seems to be a better solution in the meantime.